carserver/server_test.go

```diff
@@ -0,0 +1,147 @@
+package carserver
+
+import (
+	"bytes"
+	"context"
+	"encoding/base64"
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"testing"
+	"time"
+
+	"github.com/filecoin-project/saturn-l2/logs"
+
+	cid "github.com/ipfs/go-cid"
+
+	"github.com/libp2p/go-libp2p-core/host"
+
+	"github.com/filecoin-project/saturn-l2/testutils"
+
+	p2phttp "github.com/libp2p/go-libp2p-http"
+
+	"github.com/google/uuid"
+
+	"github.com/filecoin-project/saturn-l2/carstore"
+
+	"github.com/ipld/go-ipld-prime/codec/dagcbor"
+
+	selectorparse "github.com/ipld/go-ipld-prime/traversal/selector/parse"
+
+	"github.com/libp2p/go-libp2p-core/peer"
+
+	mocknet "github.com/libp2p/go-libp2p/p2p/net/mock"
+
+	"github.com/stretchr/testify/require"
+)
+
+func TestSimpleTransfer(t *testing.T) {
+	path := "../testdata/files/sample-v1.car"
+	temp := t.TempDir()
+	ctx := context.Background()
+
+	// create the getway api with a test http server
+	root, contents, svc := testutils.GetTestServerFor(t, path)
+	defer svc.Close()
+	gwAPI := carstore.NewGatewayAPI(svc.URL)
+	lg := logs.NewSaturnLogger()
+	cfg := carstore.Config{}
+	cs, err := carstore.New(temp, gwAPI, cfg, lg)
```

Comment:
When running inside Filecoin Station, we need Saturn L2 node to store all cache files in the directory provided by the Station. (Typically, this directory is OS-specific. See https://github.com/filecoin-project/saturn-l2/issues/36 and https://github.com/ipfs/ipfs-desktop/issues/1656 for more details.)

I am proposing to introduce a new env var, e.g. `CACHE_DIR` or `SATURN_CACHEDIR`:
- Filecoin Station sets `CACHE_DIR` env var, similarly to how it sets `FIL_WALLET_ADDRESS`
- Saturn L2 Node uses this value as the root dir for all caches

To make it easier to run saturn-l2 on its own, we can introduce a sensible default when `CACHE_DIR` is not provided. I don't think `t.TempDir()` is a good one - IMHO, we should preserve cached data across node restarts?

Reply:
Hey I have introduced an ENV VAR called ROOT_DIR.

Reply:
I haven't added a sensible default yet but we can add it down the line if need be. Should be easy.